frame/staking/src/pallet/impls.rs

```diff
@@ -1263,9 +1257,18 @@ where
 					}
 				} else {
 					// Defer to end of some `slash_defer_duration` from now.
-					<Self as Store>::UnappliedSlashes::mutate(active_era, move |for_later| {
-						for_later.push(unapplied)
-					});
+					log!(
+						debug,
+						"deferring slash of {:?}% happened in {:?} (reported in {:?}) to {:?}",
+						slash_fraction,
+						slash_era,
+						active_era,
+						slash_era + slash_defer_duration + 1,
```

Comment:
just to be sure: we strongly rely on the fact that `apply_unapplied_slashes` will be called before executing extrinsics in block at height `slash_era + slash_defer_duration + 1`, right? in particular, any unbonding / withdrawing will be processed _after_ slashing

Reply:
exactly.